File: 4-medianTwoSortedArrays/JS/medianTwoSortedArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
  let size = nums1.length + nums2.length;
  let sorted = [...nums1, ...nums2].sort(function (a, b) {
    return a - b;
  });
  half = size / 2;

  if (Number.isInteger(half)) {
    return (sorted[half - 1] + sorted[half]) / 2;
  }
  return sorted[Math.floor(half)];
};

module.exports = findMedianSortedArrays;

File: 4-medianTwoSortedArrays/Python3/medianTwoSortedArrays.py
import math

class Solution:
    def findMedianSortedArrays(self, nums1: list[int], nums2: list[int]) -> float:
        ac = nums1 + nums2
        ac.sort()
        mid = len(ac)/2.0
        if not mid.is_integer():
            return ac[math.floor(mid)]
        return (ac[int(mid-1)] + ac[int(mid)])/2.0
